File: RuneSharper.Shared/Models/ActivitiesModel.cs
using RuneSharper.Shared.Entities.Snapshots;
using RuneSharper.Shared.Extensions;

namespace RuneSharper.Shared.Models;

public class ActivitiesModel
{
    public ActivitiesModel(IEnumerable<ActivitySnapshot> activities)
    {
        foreach (var activity in activities)
        {
            var model = new ActivityModel
            {
                Score = activity.Score,
                Name = activity.Type.ToString(),
                Rank = activity.Rank
            };

            if (activity.Type.IsBoss())
                Bosses.Add(model);

            if (activity.Type.IsClue())
                Clues.Add(model);

            if (activity.Type.IsOther())
                Other.Add(model);
        }
    }

    public ICollection<ActivityModel> Clues { get; set; } = new List<ActivityModel>();
    public ICollection<ActivityModel> Bosses { get; set; } = new List<ActivityModel>();
    public ICollection<ActivityModel> Other { get; set; } = new List<ActivityModel>();
}

File: src/RuneSharper.Client/src/app/models/activities.model.ts
import { Frequency } from "./frequency";

export interface Activities {
    clues: Activity[];
    bosses: Activity[];
    other: Activity[];
}

export interface Activity {
    name: string;
    score: number;
    rank: number;
}

export interface ActivitiesChange {
    frequency: Frequency;
    dateRange: { DateFrom: Date, DateTo: Date };
    model: Activities;
}
